packages/compile-solidity/src/run.ts

```diff
@@ -412,55 +480,58 @@ function processContracts({
             contents: source
           }
-        }) => ({
-          contractName,
-          abi: orderABI({ abi, contractName, ast }),
-          metadata,
-          devdoc,
-          userdoc,
-          sourcePath: originalSourcePaths[transformedSourcePath],
-          source,
-          sourceMap,
-          deployedSourceMap: (deployedBytecodeInfo || {}).sourceMap,
-          ast,
-          legacyAST,
-          bytecode: zeroLinkReferences({
-            bytes: bytecode,
-            linkReferences: formatLinkReferences(linkReferences)
-          }),
-          deployedBytecode: zeroLinkReferences({
-            bytes: (deployedBytecodeInfo || {}).object,
-            linkReferences: formatLinkReferences(
-              (deployedBytecodeInfo || {}).linkReferences
-            )
-          }),
-          immutableReferences: (deployedBytecodeInfo || {}).immutableReferences,
-          //ideally immutable references would be part of the deployedBytecode object,
-          //but compatibility makes that impossible
-          generatedSources,
-          deployedGeneratedSources: (deployedBytecodeInfo || {})
-            .generatedSources,
-          compiler: {
-            name: "solc",
-            version: solcVersion
-          }
-        })
+        }) => {
+          return {
+            contractName,
+            abi: orderABI({ abi, contractName, ast }),
+            metadata,
+            devdoc,
+            userdoc,
+            sourcePath: originalSourcePaths[transformedSourcePath],
+            source,
+            sourceMap,
+            deployedSourceMap: (deployedBytecodeInfo || {}).sourceMap,
+            ast,
+            legacyAST,
+            bytecode: zeroLinkReferences({
+              bytes: bytecode,
+              linkReferences: formatLinkReferences(linkReferences)
+            }),
+            deployedBytecode: zeroLinkReferences({
+              bytes: (deployedBytecodeInfo || {}).object,
+              linkReferences: formatLinkReferences(
+                (deployedBytecodeInfo || {}).linkReferences
+              )
+            }),
+            immutableReferences: (deployedBytecodeInfo || {})
+              .immutableReferences,
+            //ideally immutable references would be part of the deployedBytecode object,
+            //but compatibility makes that impossible
+            generatedSources,
+            deployedGeneratedSources: (deployedBytecodeInfo || {})
+              .generatedSources,
+            compiler: {
+              name: "solc",
+              version: solcVersion
+            }
+          };
+        }
       )
   );
 }
 
-function repairOldContracts(contracts) {
-  const contractNames = [].concat(
-    ...Object.values(contracts).map(source => Object.keys(source))
-  );
+function repairOldContracts(contracts: Contracts): Contracts {
+  const contractNames = Object.values(contracts)
+    .map(source => Object.keys(source))
+    .flat();
```

Comment:
```suggestion
    .mapFlat(source => Object.keys(source));
```

Reply:
That would be nice but I didn't implement that because `flatMap` isn't supported in Node 12. I assume that ends up in the built code.

Reply:
Pretty sure it down-compiles to a call to `map(...).flat()` thanks to the target being `ES2016`, but to be sure you'd have to check the JS output. That said, it's not really worth the extra time.